src/Og.php

```diff
@@ -96,21 +102,35 @@ public static function createField($plugin_id, $entity_type, $bundle, array $set
         'targetEntityType' => $entity_type,
         'bundle' => $bundle,
         'mode' => 'default',
-        'status' => TRUE,
       ]);
     }
 
-    $widget = $form_display->getComponent($plugin_id);
-    $widget['type'] = 'og_complex';
-    $widget['settings'] = [
-      'match_operator' => 'CONTAINS',
-      'size' => 60,
-      'placeholder' => '',
-    ];
+    $form_display_definition = $og_field->getFormDisplayDefinition($settings['form_display']);
+
 
-    $form_display->setComponent($plugin_id, $widget);
+    $form_display->setComponent($plugin_id, $form_display_definition);
     $form_display->save();
 
+
+    // Set the view display for the "default" and "teaser" types.
+    $view_display_definition = $og_field->getViewDisplayDefinition($settings['view_display']);
+
+    foreach (['default', 'teaser'] as $mode) {
+
+      /** @var EntityDisplayInterface $view_display */
+      if (!$view_display = \Drupal::entityTypeManager()->getStorage('entity_view_display')->load($entity_type . '.' . $bundle . '.' . $mode)) {
```

Comment:
@pfrenssen @damiankloip I'm fighting this a bit, and I wonder if any of you folks see something I don't:

When I  create an audience field on article (e.g. `\Drupal\og\Og::createField('og_group_ref', 'node', 'article');`) the field appears as hidden in the view displays (`admin/structure/types/manage/article/display`), and in fact cannot even be dragged up to be visible.

![manage_display___site-install](https://cloud.githubusercontent.com/assets/125707/14188794/efc699c0-f792-11e5-91b8-e6991b50d4f3.jpg)

Trying to get the component returns `NULL` even-though we are saving the component here. 

```php
    /** @var EntityDisplayInterface $view_display */
    $view_display = \Drupal::entityTypeManager()->getStorage('entity_view_display')->load('node.article.default');

    dpm($view_display->getComponent('og_group_ref'));
```

What am I missing?




```